### ISSUE TITLE
union got syntax error 

### ISSUE THREAD

**Issue by aytrack** [pingcap/tidb#28053]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
select 1 union ((select 1));
```

### 2. What did you expect to see? (Required)
```sql
MySQL > select 1 union ((select 1));

+---+
| 1 |
+---+
| 1 |
+---+

```
### 3. What did you see instead (Required)
release-5.0(v5.0.3) & release-5.1(v5.1.1) execute correctly 
release-5.2(v5.2.0, v5.2.1） got syntax error.
### 4. What is your TiDB version? (Required)
v5.2.0, v5.2.1, master(6ebfe8aa4257018b5ee397d)

**Comment by ywqzzy** [comment]
This query contains subquery with multiple brackets，but the parser did not match this pattern. The following pr introduce a new pattern, and solved the problem. https://github.com/pingcap/tidb/pull/29656

**Comment by zanmato1984** [comment]
Changing labels.

**Comment by zanmato1984** [comment]
Also, fixed by https://github.com/pingcap/tidb/pull/29656 since 6.0.

**Comment by bb7133** [comment]
Close this issue since it has been fixed in master/5.4/6.0, it's up to the backport strategy whether it will be fixed in old versions.
